gradle-xmake-plugin/src/main/resources/lua/install_artifacts.lua

```diff
@@ -143,49 +139,55 @@ end
 
 -- clean artifacts
 function _clean_artifacts(installdir, targets, arch)
+    assert(xmake.version():satisfies(">= 2.9.5"), "please update xmake to >= 2.9.5")
+    for _, target in ipairs(targets) do
+        uninstall(target, {installdir = installdir,  libdir = arch})
+    end
 
-    -- append arch sub-directory
-    installdir = path.join(installdir, arch)
+    local installdir = path.join(installdir, arch)
 
-    -- clean targets artifacts in the install directory
-    for _, target in ipairs(targets) do
-        os.tryrm(path.join(installdir, path.filename(target:targetfile())))
-        if os.emptydir(installdir) then
-            os.tryrm(installdir)
+    -- clean cxxstl
+    local ndk_cxxstl = get_config("ndk_cxxstl")
+    if ndk_cxxstl then
+        local cxxstl_filename
+        if ndk_cxxstl == "c++_shared" then
+            cxxstl_filename = "libc++_shared.so"
+        elseif ndk_cxxstl == "gnustl_shared" then
+            cxxstl_filename = "libgnustl_shared.so"
+        elseif ndk_cxxstl == "stlport_shared" then
+            cxxstl_filename = "libstlport_shared.so"
         end
+        os.tryrm(installdir)
+    end
+
+    if os.emptydir(installdir) then
+        os.rmdir(installdir)
     end
 end
 
 -- main entry
-function main(libsdir, installdir, archs, ...)
+function main(installdir, arch, clean, ...)
```

Comment:
We can improve to parse arguments.

like this:

https://github.com/xmake-io/xmake/blob/124e091f949eb35a47129086f8320eb8ae5bfcbd/xmake/modules/private/utils/bin2c.lua#L25

https://github.com/xmake-io/xmake/blob/124e091f949eb35a47129086f8320eb8ae5bfcbd/xmake/modules/private/utils/bin2c.lua#L100-L106

Reply:
then `xmake l install_artfacts.lua --installdir=xxx --arch=xxx --clean`